keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/artifact/ImageHandler.kt

```diff
@@ -36,24 +37,32 @@ class ImageHandler(
           ArtifactCheckSkipped(artifact.type, artifact.name, "ActuationInProgress")
         )
       } else {
-        val latestVersion = artifact.findLatestArtifactVersion()
-        val latestBaseImageVersion = artifact.findLatestBaseAmiVersion()
+        val latestArtifactVersion = artifact.findLatestArtifactVersion()
+        val latestBaseAmiVersion = artifact.findLatestBaseAmiVersion()
         val image = artifact.findLatestAmi()
 
         if (image == null) {
           log.debug("No image found for {}", artifact.name)
         } else {
-          log.debug("Latest known {} version: {}", artifact.name, latestVersion)
+          log.debug("Latest known {} version: {}", artifact.name, latestArtifactVersion)
           log.debug("Latest baked {} version: {}", artifact.name, image.appVersion)
-          log.debug("Latest {} {} base AMI version: {}", artifact.vmOptions.baseOs, artifact.vmOptions.baseLabel, latestBaseImageVersion)
+          log.debug("Latest {} {} base AMI version: {}", artifact.vmOptions.baseOs, artifact.vmOptions.baseLabel, latestBaseAmiVersion)
           log.debug("Latest baked {} base AMI version: {}", artifact.name, image.baseAmiVersion)
         }
 
-        val versionsDiffer = image?.appVersion != latestVersion || image?.baseAmiVersion != latestBaseImageVersion
+        val versionsDiffer = image?.appVersion != latestArtifactVersion || image?.baseAmiVersion != latestBaseAmiVersion
         val regionsDiffer = !(image?.regions ?: emptySet()).containsAll(artifact.vmOptions.regions)
         if (image == null || versionsDiffer || regionsDiffer) {
-          log.info("Versions or regions for {} differ, launching bake", artifact.name)
-          launchBake(artifact, latestVersion)
+          if (bakeHistory.contains(latestArtifactVersion, latestBaseAmiVersion)) {
+            log.warn("Artifact version {} and base AMI version {} were baked previously", latestArtifactVersion, latestBaseAmiVersion)
+            publisher.publishEvent(RecurrentBakeDetected(latestArtifactVersion, latestBaseAmiVersion))
+          } else {
+            log.info("Versions or regions for {} differ, launching bake", artifact.name)
```

Comment:
can you make this log message tell which of the two differ?

Reply:
Actually, could you just use the differ's `toJson` view to print the actual difference? I know we're not using it here, but it might help us get to the bottom of why we re-bake. As I've been debugging this week I've really wanted to know what we expect vs what exists.

Reply:
I was just thinking the same thing